1261/fixes serialization of tabular data profile

shashigharti opened the pull request:
- When we set profiles as ["tabular-data-resource", "data-resource", "Other"], it is not handled in the code (line 284, metadata.py). It reads it as file path and tries to open the file and error occurs. To fix that, this fix adds the condition, which checks for the profile if it is path, remotefile or a dict.
- fixes #1261

---

Please make sure that all the checks pass. Please add here any additional information regarding this pull request. It's highly recommended that you link this PR to an issue (please create one if it doesn't exist for this PR)

shashigharti committed:
Fixed serialization error of descriptor with profile

shashigharti committed:
Changed resource.py file

shashigharti committed:
Merge branch 'main' into 1261/fixes-serialization-of-tabular-data-profile

shashigharti commented:
@roll this is ready for the review

roll reviewed (changes requested):
Hi @shashigharti,

We need to handle compatibility between v4 and v5 in:
- IMPORT: `package.metadata_transform` function (see `resource.metadata_transform`)
- EXPORT: `package.metadata_export` (not related to this issue)

But for the Package the problem is bigger. Profiles like `data-package` and `tabular-data-package` are basically useless with a new architecture. I think the only thing we can do is:
- to ignore/remove `data-package` profile in `package.metadata_transform`
- set all the resource to `type=table` if a profile is `tabular-data-package`

shashigharti committed:
Merge branch 'main' into 1261/fixes-serialization-of-tabular-data-profile

shashigharti committed:
Fixed the flow based on the feedback

Ignore "data-package"
Set resource type="table" if profile is "tabular-data-package"

shashigharti committed:
Temporary fix for type error

shashigharti commented:
> * to ignore/remove `data-package` profile in `package.metadata_transform`
> * set all the resource to `type=table` if a profile is `tabu`

@roll  Thank you!. I have made the changes accordingly and updated the PR. It is ready for review.

roll committed:
Merge branch 'main' into 1261/fixes-serialization-of-tabular-data-profile
